app/models/permalink.rb

```diff
@@ -98,6 +113,24 @@ def encode_url
   def relative_external_url
     external_url.match?(%r{\A/[^/]}) ? "#{Discourse.base_path}#{external_url}" : external_url
   end
+
+  def clear_associations
```

Comment:
If this is run before `set_association_value` every time, you could just set them all to nil without these checks right?